python/pyspark/ml/param/__init__.py

```diff
@@ -32,13 +35,17 @@ class Param(object):
     .. versionadded:: 1.3.0
     """
 
-    def __init__(self, parent, name, doc, expectedType=None):
+    def __init__(self, parent, name, doc, expectedType=None, typeConverter=None):
         if not isinstance(parent, Identifiable):
             raise TypeError("Parent must be an Identifiable but got type %s." % type(parent))
         self.parent = parent.uid
         self.name = str(name)
         self.doc = str(doc)
         self.expectedType = expectedType
+        if expectedType is not None:
+            warnings.warn("expectedType is deprecated and will be removed in 2.1.0, " +
+                          "use typeConverter instead.")
```

Comment:
"use typeConverter instead, as a keyword argument"

Also, I'd put this same message in the docstring too.